nshutijonathan/Auto-mart3
# #166546064 Build out seller create car sale advert

#### What does this PR do?
Implemented car advert post endpoint
#### Description of Task to be completed?
- Created cars tables in the database
- Added seller functionality to create car sale advert
- added user functionality to change password endpoint 
- validated the inputs of the car sale advert endpoint 
- Protected the route with the token 
#### Have The following endpoint working 
*  POST/localhost:3000/api/v2/auth/signup
*  POST/localhost:3000/api/v2/auth/sign-in
*  POST/localhost:3000/api/v2/auth/signup/admin
*  GET/localhost:3000/api/v2/users
*  GET/localhost:3000/api/v2/users/<:id>
*  DELETE/localhost:3000/api/v2/users/<:id>
*  POST/localhost:3000/api/v2/car
*  GET/localhost:3000/api/v2/cars
*  GET/localhost:3000/api/v2/users/me
*  PUT/localhost:3000/api/v2/<:email>/reset_password
#### How should this be manually tested?
* Download or clone the tutorial project code from [this repo](https://github.com/nshutijonathan/Auto-Mart3), Install all required NPM  packages by running NPM install from the command line in the project root folder (where the package.json is located).
* Start the API by running NPM  start from the command line in the project root folder, you should see the message Server listening on port 3000. You can test the API directly using an application such as Postman or you can test it with one of the single pages in the browser.
* You must have node environment configured  and .env file in the root folder
* to test authentication and authorization, add this to the header 
Key: :x-auth-token,value:JWT TOKEN
#### Any background context you want to provide?
* There is Implementation of Javascript(ES6) logic with SQL database to persist some data
#### What are the relevant pivotal tracker stories?
[#166546064](https://www.pivotaltracker.com/n/projects/2346053/stories/166546064)

#### Screenshots

![car_sale_advert3](https://user-images.githubusercontent.com/16412410/59560636-be201f80-9015-11e9-8593-b066f40442f4.PNG)

####Questions
--
